app/src/main/java/io/github/fate_grand_automata/accessibility/AccessibilityGestures.kt

```diff
@@ -162,5 +164,45 @@ class AccessibilityGestures @Inject constructor(
         TapperService.instance?.dispatchGesture(gestureDesc, callback, null)
     }
 
+    @RequiresApi(Build.VERSION_CODES.O)
+    suspend fun longPress8(location: Location, duration: Duration = 2.seconds){
+        val longPressPath = Path().moveTo(location)
+
+        val longPressDuration = duration.inWholeMilliseconds
+        var gestureDelay = 0L
+        val dragReleaseDuration = 50L
+
+        Timber.d("long pressed $location started")
+
+        val lastStroke = GestureDescription.StrokeDescription(
+            longPressPath,
+            0L,
+            longPressDuration,
+            true
+        ).also {
+            performGesture(it)
+            gestureDelay += longPressDuration
+        }
+
+        lastStroke.continueStroke(
+            longPressPath,
+            gestureDelay,
+            dragReleaseDuration,
+            false
+        ).also {
+            performGesture(it)
+        }
+
+        Timber.d("long pressed $location stopped")
+    }
+
+    override fun longPress(location: Location, duration: Duration) = runBlocking {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            longPress8(location, duration)
+        } else {
+            click(location, 2)
```

Comment:
I guess this will produce weird behavior for Android 7, but I think we should drop support for Android 7 anyways.

Reply:
I don't mind dropping the android 7

Reply:
Perhaps it's a bit out of place to ask but it'd be nice to have a "Last working build" release for no-longer-supported Android versions